features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/CustomizedStepper.jsx

```diff
@@ -0,0 +1,340 @@
+import Configuration from 'Config';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        width: '90%',
+    },
+    button: {
+        color: theme.palette.secondary,
+        marginRight: theme.spacing(1),
+    },
+    gridRequirements: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        marginTop: theme.spacing(2),
+    },
+    gridSmall: {
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'center',
+        alignItems: 'center',
+        marginLeft: 0,
+        marginTop: theme.spacing(1),
+    },
+    gridEndpoint: {
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'center',
+        alignItems: 'center',
+        marginLeft: 0,
+        marginRight: theme.spacing(4.2),
+        marginTop: theme.spacing(0.5),
+    },
+    iconTrue: {
+        color: green[500],
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        marginRight: theme.spacing(1),
+    },
+    iconFalse: {
+        color: theme.palette.grey[500],
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        marginRight: theme.spacing(1),
+    },
+    arrowIcon: {
+        position: 'absolute',
+        top: '16px',
+        right: '-76px',
+        fontSize: '7.9461rem',
+        color: theme.palette.background.default,
+        zIndex: '1',
+    },
+    label: {
+        paddingLeft: '0',
+        paddingRight: '0',
+    },
+    stepper: {
+        background: theme.palette.background.default,
+        marginLeft: theme.spacing(8),
+    },
+    box: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        background: 'white',
+        padding: '2px',
+        width: '405px',
+        height: '82px',
+        borderRight: '0',
+        marginRight: '0',
+    },
+}));
+
+/**
+ * This component renders the API steps
+ *
+ */
+export default function CustomizedSteppers() {
+    const restApi = new Api();
+    const [api, updateAPI] = useAPI();
+    const classes = useStyles();
+    const isEndpointAvailable = api.endpointConfig !== null;
+    const isTierAvailable = api.policies.length !== 0;
+    const isPrototypedAvailable =
+        api.endpointConfig !== null && api.endpointConfig.implementation_status === 'prototyped';
+    const [lifecycleState, setlifecycleState] = useState([]);
+    const [isUpdating, setUpdating] = useState(false);
+
+    useEffect(() => {
+        restApi.getLcState(api.id)
+            .then((result) => {
+                setlifecycleState(result.body.state);
+            });
+    }, []);
+
+    /**
+ * Update the LifeCycle state of the API
+ *
+ */
+    function updateLCStateOfAPI() {
+        setUpdating(true);
+        const promisedUpdate = restApi.updateLcState(api.id, 'Publish');
```

Comment:
Shall we use `api.publish()` the method instead of calling REST api directly

Reply:
fixed